superset/config.py

```diff
@@ -586,11 +586,24 @@ class CeleryConfig:  # pylint: disable=too-few-public-methods
 # The directory within the bucket specified above that will
 # contain all the external tables
 CSV_TO_HIVE_UPLOAD_DIRECTORY = "EXTERNAL_HIVE_TABLES/"
+# Function that creates upload directory dynamically based on the
+# database used, user and schema provided.
+CSV_TO_HIVE_UPLOAD_DIRECTORY_FUNC: Callable[
+    ["Database", "models.User", str], Optional[str]
+] = lambda database, user, schema: CSV_TO_HIVE_UPLOAD_DIRECTORY
 
 # The namespace within hive where the tables created from
 # uploading CSVs will be stored.
+# Make it a func, e.g. /hive/warehouse/<dbname>.db/<tablename>
 UPLOADED_CSV_HIVE_NAMESPACE = None
 
+# database
+ALLOWED_USER_CSV_SCHEMA_FUNC: Callable[
+    ["Database", "models.User"], List[str]
+] = lambda database, user: [
+    UPLOADED_CSV_HIVE_NAMESPACE  # type: ignore
```

Comment:
I think we could remove the type ignore here if you define a type above too

Reply:
mypy is not smart enough to understand the if case here, will add a comment explaining the reason behind type ignore

Reply:
really? even if you type `UPLOADED_CSV_HIVE_NAMESPACE: Optional[List[str]] = None`?

Reply:
issue here is that function is Callable[["Database", "models.User"], List[str] but UPLOADED_CSV_HIVE_NAMESPACE is optional, lambda has if case to ensure that result is always a list, and mypy is failing to determine that.